### PR TITLE
Fix duplicate upload + add scroll remembering

### PR DESCRIPTION
**Pull request by ildyria**
- improve code style on `ref()`
- use boolean globally defined in a store state for the most important modals (new album, upload) 

See in particular:
https://github.com/LycheeOrg/Lychee/pull/2693/files#diff-5585be2db0c41adcd345e27400e1476598e1979d611d3030d032d049f64f12e4
https://github.com/LycheeOrg/Lychee/pull/2693/files#diff-4fdc99a20d4bdd62b978a193d29a30934d64bf6ceb26f010bd441b9e9ac167db
TODO later: migrate the other booleans (rename etc) to that store.

### CODE REVIEW

File: resources/js/components/forms/settings/BoolField.vue
Comment (tinohager) [inline resources/js/components/forms/settings/BoolField.vue:1]: the code is duplicate `props.config.value === "1"`, maybe you could save the result above in a const

Comment (ildyria) [reply resources/js/components/forms/settings/BoolField.vue:1]: But a const would not be updated on a save of the value as done with the watch :/

File: resources/js/views/gallery-panels/Map.vue [hunk 1]
```diff
@@ -68,16 +70,16 @@ function goBack() {
...
 const camera_date = trans("lychee.CAMERA_DATE");
-const map_provider = ref(undefined) as Ref<undefined | App.Http.Resources.GalleryConfigs.MapProviderData>;
+const map_provider = ref<App.Http.Resources.GalleryConfigs.MapProviderData | undefined>(undefined);
 const map = ref(undefined) as Ref<L.Map | undefined>;
-const bounds = ref(undefined as undefined | LatLngBoundsLiteral);
-const photoLayer = ref(undefined as any);
-const trackLayer = ref(undefined as any);
-const data = ref(undefined as undefined | App.Http.Resources.Collections.PositionDataResource);
+const bounds = ref<LatLngBoundsLiteral | undefined>(undefined);
+const photoLayer = ref<any>(undefined);
```
Comment (tinohager) [inline resources/js/views/gallery-panels/Map.vue:80]: `any` is bad, is there no model that can be used?

File: resources/js/components/forms/settings/BoolField.vue
Comment (tinohager) [reply resources/js/components/forms/settings/BoolField.vue:1]: computed?

File: resources/js/views/gallery-panels/Map.vue (hunk 1 above)
Comment (ildyria) [reply resources/js/views/gallery-panels/Map.vue:80]: Yeah I know, but this is untyped code from a library. On that one not much chance, and I don't feel like typing the damn thing. :')

**Review by tinohager — APPROVED** [review]

**Review by d7415 — APPROVED** [review]

**Comment by tinohager** [comment]
When does the pull-request flow into the alpha version of the docker container?

**Comment by ildyria** [comment]
@tinohager about tonight it will be available on `nightly`/`dev` and `alpha`. :)

**Pull request merged**